src/olympia/scanners/models.py

```diff
@@ -24,6 +43,7 @@ class ScannerResult(ModelBase):
         null=True,
     )
     has_matches = models.NullBooleanField()
+    matched_rules = models.ManyToManyField(ScannerRule)
```

Comment:
It might be a good idea to start having an intermediary model right away, even if we don't use it yet. It'll make adding fields/custom logic easier in the future.